pyinfra/api/connectors/ssh.py

```diff
@@ -112,50 +110,20 @@ def _get_private_key(state, key_filename, key_password):
     return key
 
 
-def get_ssh_config(hostname):
-    ssh_config = SSHConfig()
-    user_config_file = path.expanduser('~/.ssh/config')
-    cfg = dict()
-
-    if path.exists(user_config_file):
-        with open(user_config_file) as f:
-            ssh_config.parse(f)
-            host_config = ssh_config.lookup(hostname)
-            if 'user' in host_config:
-                cfg['username'] = host_config['user']
-
-            if 'proxycommand' in host_config:
-                cfg['sock'] = ProxyCommand(host_config['proxycommand'])
-
-            if 'identityfile' in host_config:
-                cfg['pkey'] = _get_private_key(host_config['identityfile'])
```

Comment:
How will `sshuserclient` work with password-encrypted SSH keys? Currently in CLI mode this will ask the user for any key password but this will be lost by making the switch.

Reply:
doesn't everybody uses ssh-agent or similar nowadays?

Reply:
thanks for spotting the key policy setting in sshuerclient, I will remove it.

Reply:
sshuserclient updated.

Reply:
I adapted the test a bit hastily, I am not sure if you intend to override paramiko default behavior of look_for_keys and allow_agent.

Reply:
@tobald yes that's correct. The solution would be just to have:

```
kwargs = {
    'allow_agent': False,
    'look_for_keys': False,
}
...
```

This this should correctly behave and respect any `host.data.X` and CLI arguments, while also satisfying the `ssh_config` defaults :)